uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/ui/screens/habits/show/views/HistoryCard.kt

```diff
@@ -105,12 +107,19 @@ class HistoryCardPresenter(
             val oldest = habit.computedEntries.getKnown().lastOrNull()?.timestamp ?: today
             val entries = habit.computedEntries.getByInterval(oldest, today)
             val series = if (habit.isNumerical) {
-                entries.map {
-                    Entry(it.timestamp, max(0, it.value))
-                }.map {
-                    when (it.value) {
-                        0 -> HistoryChart.Square.OFF
-                        else -> HistoryChart.Square.ON
+                if (habit.targetType == NumericalHabitType.AT_LEAST) {
+                    entries.map {
+                        when (max(0, it.value)) {
+                            0 -> HistoryChart.Square.OFF
+                            else -> HistoryChart.Square.ON
+                        }
+                    }
+                } else {
+                    entries.map {
+                        when {
+                            max(0.0, it.value / 1000.0) < 2 * habit.targetValue -> HistoryChart.Square.ON
+                            else -> HistoryChart.Square.OFF
+                        }
```

Comment:
For now, could you please make it `ON` if `value <= target`? I think using `value <= 2*target` will be unintuitive for most users.